### ISSUE TITLE
Incorrect usage of bytes constructor

### ISSUE THREAD

**Issue by zilm13** [protolambda/remerkleable#15]
From the python spec: 
A zero-filled bytes object of a specified length: bytes(10)
https://docs.python.org/3.10/library/stdtypes.html#binaryseq
So when initialized with int or something like `0x01` in https://github.com/protolambda/remerkleable/blob/master/remerkleable/byte_arrays.py#L28 you get `00`-filled bytes of input length. 
Broken usage is, for example, here: https://github.com/ethereum/consensus-specs/blame/dev/specs/eip4844/beacon-chain.md#L58

**Comment by zilm13** [comment]
I'm not sure, maybe we should change `consensus-specs` input but it's easy repeatable error

**Comment by protolambda** [comment]
Nice catch! I think `consensus-specs` should specify it as string input for clarity, based on previous similar usage in the withdrawal prefix constants here: https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#withdrawal-prefixes

That said, it's indeed an easy thing to get wrong, at least for the case when the value is equal to the bytevector length. (since it's checked here: https://github.com/protolambda/remerkleable/blob/94ed7e2cd143a17475155ed45c5f740af9290bb1/remerkleable/byte_arrays.py#L81

And for a `ByteList` it makes sense to just follow the `bytes` behavior. For the fixed-size `ByteVector` it doesn't make sense, as one should just call the constructor without arguments to get the default value. So to make it less error-prone, I think we should add a check to the `ByteVector` constructor, before calling the super constructor, like `if len(args) == 1 and isinstance(args[0], int): raise Exception(f"cannot init fixed-length ByteVector[{byte_len}] with dynamic-length {args[0]}")`

What do you think?

**Comment by zilm13** [comment]
I agree with you as it's better to throw an exception as it could be not one byte integer but any size and even if we do conversion back it will be useless two-way conversion with new potential for error by feeling leading zeros for example. I think we should go exception way
